Build failure: trurl@0.8 (test failure with unreleased upstream fix)

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `trurl`@0.8

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/bcq7v4whvvsmbsn9gpsvdr9lffgskphz-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: -j8 SHELL=/nix/store/zzpm4317hn2y29rm46krsasaww9wxb1k-bash-5.2-p15/bin/bash PREFIX=\$\(out\)
clang $(curl-config --cflags) -W -Wall -Wshadow -Werror -pedantic -g -std=gnu99   -c -o trurl.o trurl.c
clang trurl.o -o trurl $(curl-config --libs) 
@nix { "action": "setPhase", "phase": "checkPhase" }
running tests
check flags: -j8 SHELL=/nix/store/zzpm4317hn2y29rm46krsasaww9wxb1k-bash-5.2-p15/bin/bash PREFIX=\$\(out\) VERBOSE=y test
1: passed	example.com
2: passed	http://example.com
3: passed	https://example.com
4: passed	hp://example.com
5: passed	
6: passed	ftp.example.com
7: passed	https://example.com/../moo
8: passed	https://example.com/.././moo
9: passed	https://example.com/test/../moo
10: passed	localhost --append path=moo
11: passed	localhost -a path=moo
12: passed	--set host=moo --set scheme=http
13: passed	-s host=moo -s scheme=http
14: passed	--set host=moo --set scheme=https --set port=999
15: passed	--set host=moo --set scheme=ftps --set path=/hello
16: passed	--url https://curl.se --set host=example.com
17: passed	--set host=example.com --set scheme=ftp
18: passed	--url https://curl.se/we/are.html --redirect here.html
19: passed	--url https://curl.se/we/../are.html --set port=8080
20: passed	https://curl.se:22/ -s port=443
21: passed	https://curl.se:22/ -s port=443 --get '{url}'
22: passed	--url https://curl.se/we/are.html --get '{path}'
23: passed	--default-port --url https://curl.se/we/are.html --get '{port}'
24: passed	--url https://curl.se/we/are.html --get '{scheme}'
25: passed	--url https://hello@curl.se/we/are.html --get '{user}'
26: passed	--url https://hello:secret@curl.se/we/are.html --get '{password}'
27: passed	--url 'imap://hello:secret;crazy@curl.se/we/are.html' --get '{options}'
28: passed	--url https://curl.se/we/are.html --get '{host}'
29: passed	--url https://10.1/we/are.html --get '{host}'
30: passed	--url 'https://[fe80::0000:20c:29ff:fe9c:409b]:8080/we/are.html' --get '{host}'
31: passed	--url 'https://[fe80::0000:20c:29ff:fe9c:409b%euth0]:8080/we/are.html' --get '{zoneid}'
32: passed	--url 'https://[fe80::0000:20c:29ff:fe9c:409b%eth0]:8080/we/are.html' --get '{zoneid}'
33: passed	--url 'https://curl.se/we/are.html?user=many#more' --get '{query}'
34: passed	--url 'https://curl.se/we/are.html?user=many#more' --get '{fragment}'
35: passed	--url https://curl.se/we/are.html -g '{default:port}'
36: passed	--url https://curl.se/hello --append path=you
37: passed	--url https://curl.se/hello --append 'path=you index.html'
38: passed	--url 'https://curl.se?name=hello' --append query=search=string
39: passed	--url https://curl.se/hello --set user=:hej:
40: passed	--url https://curl.se/hello --set user=hej --set password=secret
41: passed	--url https://curl.se/hello --set query:=user=me
42: passed	--url https://curl.se/hello --set query=user=me
43: passed	--url https://curl.se/hello --set 'fragment= hello'
44: passed	--url https://curl.se/hello --set fragment:=%20hello
45: passed	localhost --append query=hello=foo
46: passed	localhost -a query=hello=foo
47: passed	'https://example.com?search=hello&utm_source=tracker' --trim 'query=utm_*'
48: passed	'https://example.com?search=hello&utm_source=tracker&more=data' --trim 'query=utm_*'
49: passed	'https://example.com?search=hello&more=data' --trim 'query=utm_*'
50: passed	'https://example.com?utm_source=tracker' --trim 'query=utm_*'
51: passed	'https://example.com?search=hello&utm_source=tracker&more=data' --trim query=utm_source
52: passed	'https://example.com?search=hello&utm_source=tracker&more=data' --trim query=utm_source --trim query=more --trim query=search
53: passed	--accept-space --url 'gopher://localhost/ with space'
54: passed	--accept-space --url 'https://localhost/?with space'
55: passed	https://daniel@curl.se:22/ -s port= -s user=
56: passed	'https://example.com?moo&search=hello' --trim query=search
57: passed	'https://example.com?search=hello&moo' --trim query=search
58: passed	'https://example.com?search=hello' --trim query=search --append query=moo
59: passed	https://hello:443/foo
60: passed	ftp://hello:21/foo
61: passed	https://hello:443/foo -s scheme=ftp
62: passed	ftp://hello:443/foo -s scheme=https
63: passed	'https://example.com?utm_source=tra%20cker&address%20=home&here=now&thisthen' -g '{query:utm_source}'
64: passed	'https://example.com?utm_source=tra%20cker&address%20=home&here=now&thisthen' -g '{:query:utm_source}'
65: passed	'https://example.com?utm_source=tra%20cker&address%20=home&here=now&thisthen' -g '{:query:utm_}'
66: passed	'https://example.com?utm_source=tra%20cker&address%20=home&here=now&thisthen' -g '{:query:UTM_SOURCE}'
67: passed	'https://example.com?utm_source=tracker&monkey=123' --sort-query
68: passed	'https://example.com?a=b&c=d&' --sort-query
69: passed	'https://example.com?a=b&c=d&' --sort-query --trim query=a
70: passed	example.com:29 --set port=
71: passed	--url HTTPS://example.com
72: passed	--url https://EXAMPLE.com
73: passed	--url https://example.com/FOO/BAR
74: passed	--url '[2001:0db8:0000:0000:0000:ff00:0042:8329]'
75: passed	'https://example.com?utm=tra%20cker:address%20=home:here=now:thisthen' --sort-query --query-separator :
76: passed	'foo?a=bCd=eCe=f' --query-separator C --trim query=d
77: passed	localhost -g '{scheme} {host'
78: passed	localhost -g '[scheme] [host'
79: passed	localhost -g '\{{scheme}\['
80: passed	localhost -g '\\['
81: passed	'https://u:s@foo?moo' -g '[scheme][user][password][query]'
82: passed	'hej?a=b&a=c&a=d&b=a' -g '{query-all:a}'
83: passed	'https://curl.se?name=mr%00smith' --get '{query:name}'
84: passed	https://curl.se --iterate 'port=80 81 443'
85: passed	https://curl.se --iterate 'port=81 443' --iterate 'scheme=sftp moo'
86: passed	https://curl.se --iterate 'port=81 443' --iterate 'scheme=sftp moo' --iterate 'port=2 1'
87: passed	https://curl.se -s host=localhost --iterate 'port=22 23'
88: passed	hello@localhost --iterate 'host=one two' -g '{host} {user}'
89: passed	'https://example.com?utm=tra%20cker&address%20=home&here=now&thisthen' --json
90: passed	'ftp://smith:secret@example.com:33/path?search=me#where' --json
91: passed	example.com --json
92: passed	example.com other.com --json
93: passed	localhost --iterate 'host=one two' --json
94: passed	--json -s scheme=irc -s host=curl.se
95: passed	--json -s host=curl.se
[91m96: failed	--verify --json ftp://example.org '' git://curl.se/
[0m--- stdout --- 
expected:
[{'url': 'ftp://example.org/', 'parts': {'scheme': 'ftp', 'host': 'example.org', 'path': '/'}}]
got:
[{'url': 'ftp://example.org/', 'parts': {'scheme': 'ftp', 'host': 'example.org', 'path': '/'}}]
--- returncode --- 
expected:
9
got:
9
--- stderr --- 
expected:
'trurl error: No host part in the URL []\ntrurl error: Try trurl -h for help\n'
got:
'trurl error: Malformed input to a URL function []\ntrurl error: Try trurl -h for help\n'

97: passed	-s scheme=imap
98: passed	-g '{query:}' 'http://localhost/?=bar'
99: passed	--json 'https://curl.se/?&&&'
100: passed	--json --trim 'query=f*' 'localhost?foo&bar=ar'
101: passed	'https://example.com?search=hello&utm_source=tracker&utm_block&testing' --trim 'query=utm_*' --json
102: passed	'https://räksmörgås.se' -g '{default:puny:url}'
103: passed	'https://räksmörgås.se' -g '{puny:url}'
104: passed	'https://räksmörgås.se' -g '{puny:host}'
105: passed	'imaps://user:password;crazy@[ff00::1234%hello]:1234/path?a=b&c=d#fragment' --json
106: passed	http://example.com/ --get 'port: {port}, default:port: {default:port}'
107: passed	http://example.com:8080/ --get 'port: {port}, default:port: {default:port}'
108: passed	localhost -s host=foo --iterate 'host=bar baz'
[91m109: failed	emanuele6://curl.se/trurl '' https://example.org
[0m--- stdout --- 
expected:
'emanuele6://curl.se/trurl\nhttps://example.org/\n'
got:
'emanuele6://curl.se/trurl\nhttps://example.org/\n'
--- returncode --- 
expected:
0
got:
0
--- stderr --- 
expected:
'trurl note: No host part in the URL []\n'
got:
'trurl note: Malformed input to a URL function []\n'

110: passed	--verify --no-guess-scheme hello
111: passed	--verify -f testfiles/test0000.txt
112: passed	-f testfiles/test0001.txt
113: passed	--no-guess-scheme foo hi https://example.org hey git://curl.se
114: passed	-f testfiles/test0002.txt --json
115: passed	--accept-space -s 'query:=x=10&x=2 3' localhost
116: passed	-s 'path=\\' -g '{path}\n{:path}' localhost
117: passed	-s 'path=\\' --json localhost
118: passed	-s 'path=\\' -g '{path}\n{:path}' --urlencode localhost
119: passed	-s 'path=abc\\' -s 'query:=a&b&a%26b' --urlencode --json localhost
120: passed	-s scheme:=http -s host:=localhost -s path:=/ABC%5C%5C -s 'query:=a&b&a%26b'
121: passed	-g '{query:b}\t{query-all:a}\n{:query:b}\t{:query-all:a}' 'https://example.org/foo?a=1&b=%23&a=%26#hello'
122: passed	--urlencode -g '{query:b}\t{query-all:a}\n{:query:b}\t{:query-all:a}' 'https://example.org/foo?a=1&b=%23&a=%26#hello'
Finished:
Failed! - Failed: 2, Passed: 120, Skipped: 0, Total: 122
make: *** [Makefile:32: test] Error 1
```

### Additional context

This was reported upstream in September as <https://github.com/curl/trurl/issues/239>. 

It looks like there's a fix available on the upstream `master` branch, however it has not yet been released:

<https://github.com/curl/trurl/pull/240>

We should probably request that they release a hotfix. There haven't been any releases since June, but there has been activity on the `master` branch since then, including bug fixes.

For the time being, maybe we just pin to the latest commit (<https://github.com/curl/trurl/commit/3819d80e5950f3ad7a3bf097b1897cb9d392c2d8>)? Looks like their CI reports success.

### Notify maintainers

@christoph-heiss 

### Metadata

(flakes tracking nixos-unstable @ 63678e9f3d3afecfeafa0acead6239cdb447574c)

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 22.6.0, macOS 13.6`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.1`
 - channels(cdom): `"home-manager"`
 - channels(root): `""`
 - nixpkgs: `/nix/store/y3fdd8m100ffn26fgkxq5nxaph9yjv3a-xjviahzwa7x51vl51kc3c1k1n1jmhpd5-source`
```

--- a/pkgs/tools/networking/trurl/default.nix
+++ b/pkgs/tools/networking/trurl/default.nix
@@ -2,14 +2,21 @@
 
 stdenv.mkDerivation rec {
   pname = "trurl";
-  version = "0.8";
+  version = "0.9";
 
   src = fetchFromGitHub {
     owner = "curl";
     repo = pname;
     rev = "${pname}-${version}";
-    hash = "sha256-KHJMxzHqHW8WbeD6jxyuzZhuHc5x4B7fP/rYAK687ac=";
+    hash = "sha256-R/7Saea84iqUICAAQz9zcfDLDud4uFedY2752xyg+oE=";
   };
+
+  # The version number was forgotten to be updated for the release,
+  # so do it manually in the meantime.
+  # See https://github.com/curl/trurl/discussions/244#discussioncomment-7436369
+  postPatch = ''
+    substituteInPlace version.h --replace 0.8 0.9
+  '';
 
   outputs = [ "out" "dev" "man" ];
   separateDebugInfo = stdenv.isLinux;

